### PR TITLE
Made doughnut wasm packages as optional dependencies and used try/cat…

### PR DESCRIPTION
**Pull request by KarishmaBothara**
Don't know the best way of doing it.. For now made the doughnut wasm packages as optional dependencies and conditional used them in doughnut class to resolve issue on react-native for Sylo.
It wasn't possible with es6 import way to conditionally use a package..
Have tested it by removing the wasm packages from node modules and tested it

### CODE REVIEW

File: packages/api/package.json [hunk 1]
```diff
@@ -33,8 +33,10 @@
   "resolution": {
     "@polkadot/keyring": "^1.6.1"
   },
+  "optionalDependencies": {
```
Comment (holygits) [inline packages/api/package.json:36]: devDeps should be ok to leave as is

File: packages/api/test/e2e/doughnut.e2e.ts [hunk 2]
```diff
@@ -23,7 +21,16 @@ import initApiPromise from '../../../../jest/initApiPromise';
 import { cryptoWaitReady } from '@plugnet/util-crypto';
 import Doughnut from '@cennznet/types/Doughnut';
 
+let DoughnutMaker, encodeCennznut;
 const [CENNZ, CPAY] = [16_000, 16_001];
+try {
```
Comment (holygits) [inline packages/api/test/e2e/doughnut.e2e.ts:26]: using an `if` check should help bundlers ignore the import vs. the try catch

File: packages/api/test/e2e/doughnut.e2e.ts [hunk 3]
```diff
@@ -13,8 +13,6 @@
 // limitations under the License.
 
 import { hexToU8a } from '@polkadot/util';
```
Comment (holygits) [inline packages/api/test/e2e/doughnut.e2e.ts:15]: I don't think this needs to change, tests shouldn't break package build on normal install?

Comment (KarishmaBothara) [reply packages/api/test/e2e/doughnut.e2e.ts:15]: ya... will revert this file

File: packages/types/src/Doughnut.ts [hunk 4]
```diff
@@ -31,7 +30,16 @@ export default class Doughnut extends Raw {
     // NOT only doughnut bytes.
     // We must ensure that any doughnut specific bytes are consumed
     // so that the buffer [[value]] may be passed along to decode other types successfully
-    const doughnut = DoughnutCodec.decode(value as Uint8Array);
-    super(registry, doughnut.encode());
+    let doughnut;
+    // @ts-ignore
+    if (DOUGHNUT_SUPPORT === true) {
```
Comment (holygits) [inline packages/types/src/Doughnut.ts:35]: can you add a comment in this if block about the conditional compilation also

File: packages/types/src/Doughnut.ts [hunk 5]
```diff
@@ -31,7 +30,17 @@ export default class Doughnut extends Raw {
...
+      // Added the above flag to support conditional compilation for exclusion/inclusion of wasm packages
+      const DoughnutCodec = require('@plugnet/doughnut-wasm');
+      doughnut = DoughnutCodec.Doughnut.decode(value as Uint8Array);
+      doughnut = doughnut.encode();
+    } else {
+      // fallback
+      doughnut = value;
```
Comment (alexsednz) [inline packages/types/src/Doughnut.ts:42]: I would prefer an error to be thrown, or the very least a log is generated under this condition.

Comment (holygits) [reply packages/types/src/Doughnut.ts:42]: fair point, maybe log and return empty Doughnut

**Pull request merged**